run/helm/snap/templates/daemonset.yaml

```diff
@@ -0,0 +1,43 @@
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: snap
+  namespace: {{ .Release.Namespace }}
+  labels:
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+spec:
+  template:
+    metadata:
+      name: snap
+      labels:
+        daemon: snapteld
+    spec:
+      containers:
+      - name: {{ .Chart.Name }}
+        image: "{{ .Values.image.repo }}:{{ .Values.image.tag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        securityContext:
+          privileged: {{ .Values.privileged }}
+        ports:
+        - containerPort: 8181
+        volumeMounts:
+        - name: autoload
+          mountPath: {{ .Values.config.autodiscover_path }}
+        - name: config
+          mountPath: /etc/snap
+        {{- range $key, $val := .Values.mounts }}
+        - name: {{ $key | replace "/" "" | replace "." "" }}
+          mountPath: {{ $val | quote }}
+        {{- end }}
+      volumes:
+      - name: autoload
+        hostPath:
```

Comment:
I agree with the sentiment that hostPath is not an entirely supported features on kubernetes. We've experienced two examples of this:

1. the issue with file system permission, containers have to run privileged and as root to write data to a hostPath (alternatively, you could manually set the permissions)
2. GCP doesn't even honor hostPath as a persistent disk. On the case of upgrades, all hostPath directories are blown away. We've lost data b/c of this before.

I don't think that solutions should assume they can keep data on hostPaths. If data needs to survive a container restart, it should be stored on a persistent external volume.

We've been bitten

In the case of GCP, when kubernetes are installed